src/routes/conversation/[id]/+page.svelte

```diff
@@ -21,21 +21,83 @@
 	import { addChildren } from "$lib/utils/tree/addChildren";
 	import { addSibling } from "$lib/utils/tree/addSibling";
 	import { fetchMessageUpdates } from "$lib/utils/messageUpdates";
-	import { createConvTreeStore } from "$lib/stores/convTree";
 	import type { v4 } from "uuid";
 	import { useSettingsStore } from "$lib/stores/settings.js";
+	import { browser } from "$app/environment";
 
 	export let data;
 
 	$: ({ messages } = data);
 
 	let loading = false;
 	let pending = false;
+	let initialRun = true;
 
 	$: activeModel = findCurrentModel([...data.models, ...data.oldModels], data.model);
 
 	let files: File[] = [];
 
+	// create a linear list of `messagesPath` from `messages` that is a tree of threaded messages
+	$: messagesPath = createMessagesPath(messages);
+	$: messagesAlternatives = createMessagesAlternatives(messages);
+
+	$: if (browser && messagesPath.at(-1)?.id) {
+		localStorage.setItem("leafId", messagesPath.at(-1)?.id as string);
+	}
+
+	function createMessagesPath(messages: Message[], msgId?: Message["id"]): Message[] {
+		if (initialRun) {
+			if (!msgId && $page.url.searchParams.get("leadId")) {
+				msgId = $page.url.searchParams.get("leadId") as string;
+				$page.url.searchParams.delete("leadId");
```

Comment:
```suggestion
			if (!msgId && $page.url.searchParams.get("leafId")) {
				msgId = $page.url.searchParams.get("leafId") as string;
				$page.url.searchParams.delete("leafId");
```

Lil typo there I think